MekHQ/src/mekhq/campaign/personnel/education/EducationController.java

```diff
@@ -1046,7 +1046,9 @@ private static boolean graduateAdult(Campaign campaign, Person person, Academy a
      * @param resources  the resource bundle containing localized strings
      */
     private static void reportMastersOrDoctorateGain(Campaign campaign, Person person, Academy academy, ResourceBundle resources) {
-        if (person.getEduHighestEducation().isPostGraduate()) {
+        int educationLevel = academy.getEducationLevel(person) - 1; // we reduce by 1 to factor in the increase level from graduating
+
+        if (educationLevel == 3) {
```

Comment:
Let's document these magic numbers or eliminate their usage

Reply:
To be honest that whole block of code wasn't the best written. I went ahead and refactored it, incorporating some color changes I'm going to push elsewhere (to avoid having to deal with conflicts).  #4515 for more details.